parsl/dataflow/dflow.py

```diff
@@ -717,3 +718,58 @@ def load_checkpoints(self, checkpointDirs):
             raise BadCheckpoint("checkpointDirs expects a list of checkpoints")
 
         return self._load_checkpoints(checkpointDirs)
+
+
+class DataFlowKernelLoader(object):
+    """Manage which DataFlowKernel is active.
+
+    This is a singleton class containing only class methods. You should not need to instantiate this class.
+    """
+
+    _dfk = None
+
+    @classmethod
+    def set_default(cls, payload, **kwargs):
+        """If there is no active DataFlowKernel, load one.
+
+        Args:
+            - payload (str or DataFlowKernel) : Payload to load in case there is no active DataFlowKernel.
+              If a string is passed, it should be the path to a file which defines a dictionary named `config`;
+              this will be passed to the DataFlowKernel constructor.
+            - **kwargs : Arbitrary keyword arguments which are passed to the DataFlowKernel constructor.
+
+        """
+        if cls._dfk is None:
```

Comment:
Should we return a bool, so that whoever calls can know if this succeeded ?

Reply:
I don't follow. If it doesn't succeed, it will raise an error, which the caller can handle. That seems cleaner to me than checking against a bool. In the former case, the error-handling code would only need to be traversed if there's an error, whereas a check against a bool would have to be traversed every time.